microprofile/fault-tolerance-smallrye/extension/src/main/java/org/wildfly/extension/microprofile/faulttolerance/deployment/MicroProfileFaultToleranceDeploymentProcessor.java

```diff
@@ -0,0 +1,96 @@
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.wildfly.extension.microprofile.faulttolerance.deployment;
+
+import static org.jboss.as.weld.Capabilities.WELD_CAPABILITY_NAME;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import com.netflix.config.ConfigurationManager;
+import org.apache.commons.configuration.AbstractConfiguration;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.as.controller.capability.CapabilityServiceSupport;
+import org.jboss.as.server.deployment.Attachments;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.as.server.deployment.DeploymentUnit;
+import org.jboss.as.server.deployment.DeploymentUnitProcessingException;
+import org.jboss.as.server.deployment.DeploymentUnitProcessor;
+import org.jboss.as.weld.Capabilities;
+import org.jboss.as.weld.WeldCapability;
+import org.jboss.modules.Module;
+import org.wildfly.extension.microprofile.faulttolerance.MicroProfileFaultToleranceLogger;
+
+/**
+ * @author Radoslav Husar
+ */
+public class MicroProfileFaultToleranceDeploymentProcessor implements DeploymentUnitProcessor {
+
+    private static AtomicBoolean hystrixConfigured = new AtomicBoolean();
+
+    @Override
+    public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitProcessingException {
+        DeploymentUnit deploymentUnit = phaseContext.getDeploymentUnit();
+
+        if (!hystrixConfigured.getAndSet(true)) {
+            // TODO Alternative solution to support hystrix configuration only from the "globally scoped config"
+            // as opposed to deployments providing their configuration.
+            //ClassLoader extensionClassLoader = MicroProfileFaultToleranceExtension.class.getClassLoader();
+            //Config mpConfig = ConfigProvider.getConfig(extensionClassLoader);
+
+            Module module = deploymentUnit.getAttachment(Attachments.MODULE);
+            Config mpConfig = ConfigProvider.getConfig(module.getClassLoader());
+            AbstractConfiguration configInstance = ConfigurationManager.getConfigInstance();
+
+            // We need to iterate over all keys since the key names are dynamic and not known in advance
+            for (String key : mpConfig.getPropertyNames()) {
+                if (key.startsWith("hystrix.")) {
+                    String value = mpConfig.getValue(key, String.class);
+                    MicroProfileFaultToleranceLogger.ROOT_LOGGER.debugf("Configuring Hystrix: %s=%s", key, value);
+                    configInstance.setProperty(key, value);
+                }
+            }
+        } else {
+            MicroProfileFaultToleranceLogger.ROOT_LOGGER.hystrixAlreadyConfigured(deploymentUnit.getName());
+        }
+
+
+        // Weld Extension
+        CapabilityServiceSupport support = deploymentUnit.getAttachment(Attachments.CAPABILITY_SERVICE_SUPPORT);
```

Comment:
I think the new microprofile-fault-tolerance subsystem could need to create a capability with a hard requirement on Weld capability. 

If CDI is not a hard requirement to make microprofile-fault-tolerance subsystem work properly, then I think at a minimum, this piece of code getting the Weld Capability should be moved to be processed before L:55. At first glance, it sounds like a possible bug if the error on L:85 is thrown after processing L:55 when we have multiple deployments being deployed at the same time.

Reply:
The weld code is similar to MP Config IIRC, if its wrong, its wrong across the board. Also, switched the code blocks - you are right.